### ISSUE TITLE
[BUG] Mod incorrectly identified as sever modlist mismatch

### ISSUE THREAD

**Issue by micle546** [Nolij/Zume#29]
Using Monifactory 0.9.10 on server and client, with no additional modifications.

`[16Nov2024 12:47:08.402] [Netty Client IO #4/DEBUG] [net.minecraftforge.client.ForgeHooksClient/CLIENTHOOKS]: Received FML ping data from server at 10.0.1.228:24477: FMLNETVER=3, mod list is compatible : false, channel list is compatible: true, extra server mods: {}
[16Nov2024 12:47:08.402] [Netty Client IO #4/INFO] [net.minecraftforge.client.ForgeHooksClient/CLIENTHOOKS]: Client has mods that are missing on server: [zume]`

But as stated here, 
<img width="625" alt="image" src="https://github.com/user-attachments/assets/825aa9a4-a04a-44e3-9b40-2d9110b3d74e">

It should not be used on serverside.

afaik, there are some mods that have a fml flag that let fml know to exclude it from the server modlist compatibility check.

Thanks!

Don't know if this helps, but:
https://docs.minecraftforge.net/en/1.12.x/gettingstarted/structuring/#what-is-mod

**Comment by Nolij** [comment]
Not sure if it addressed the symptom or not, but I added the `clientSideOnly` suggestion in [this commit](https://github.com/Nolij/Zume/commit/1cf633ef089a652868f517d25aad6700bf450825). Please test [2.0.0-dev.1](https://github.com/Nolij/Zume/releases/tag/release%2F2.0.0-dev.1) to see if this issue still happens.

**Comment by micle546** [comment]
Won't start. As it's 4am I'll have to remember to look more later

[crash-2024-11-30_04.13.00-fml.txt](https://github.com/user-attachments/files/17964852/crash-2024-11-30_04.13.00-fml.txt)

**Comment by Nolij** [comment]
Yeah turns out I broke things. Will fix that soon and post another link here when that's done.

**Comment by Nolij** [comment]
Try [2.0.0-dev.2](https://github.com/Nolij/Zume/releases/tag/release%2F2.0.0-dev.2)
